File: app/src/main/java/com/huanchengfly/tieba/post/models/database/SearchHistory.java
package com.huanchengfly.tieba.post.models.database;

import org.litepal.crud.LitePalSupport;

public class SearchHistory extends LitePalSupport {
    private int id;
    private long timestamp;
    private String content;

    public SearchHistory(String content) {
        this.timestamp = System.currentTimeMillis();
        this.content = content;
    }

    public int getId() {
        return id;
    }

    public long getTimestamp() {
        return timestamp;
    }

    public void setTimestamp(long timestamp) {
        this.timestamp = timestamp;
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }
}

File: app/src/main/java/com/huanchengfly/tieba/post/models/database/SearchHistory.kt
package com.huanchengfly.tieba.post.models.database

import androidx.compose.runtime.Immutable
import org.litepal.crud.LitePalSupport

@Immutable
data class SearchHistory(
    val content: String,
    val timestamp: Long = System.currentTimeMillis(),
) : LitePalSupport() {
    val id: Long = 0
}
